File: server/response_writer.py
import socket

from model import Response
from response_payload_writer import ResponsePayloadWriter
from socket_helper import SockHelper


class ResponseWriter:
    def __init__(self, sock: socket.socket):
        self.__sock = sock
        self.__helper = SockHelper(self.__sock)

    def write_response(self, resp: Response):
        server_version = resp.get_version()
        self.__helper.write_byte(server_version)
        response_code = resp.get_response_code()
        self.__helper.write_short(response_code)
        ResponsePayloadWriter(resp, self.__sock).write_payload()

File: client/ResponseWriter.cpp
/*
 * ResponseWriter.cpp
 *
 *  Created on: 27 Dec 2020
 *      Author: Roy Dar
 */

#include "ResponseWriter.h"
#include "CommunicationSerilizationHelper.h"

ResponseWriter::ResponseWriter(boost::asio::ip::tcp::socket *socket)
{
	this->socket = socket;
}

ResponseWriter::~ResponseWriter()
{
}

void ResponseWriter::writeResponse(Response *resp)
{
	CommunicationSerilizationHelper helper(socket);
	helper.writeByte(resp->getVersion());
	helper.writeShort(resp->getStatus());

	if (resp->getStatus() < 1002)
	{
		helper.writeStr(resp->getFileName());
	}

	if (resp->getStatus() < 212)
	{
		helper.writeBytes(resp->getPayload());
	}
}
